File: src/r3_pattern2/Main.java
package r3_pattern2;

import java.io.BufferedReader;
import java.io.InputStreamReader;
/**
 * @author Pranshu Aggarwal
 * @problem https://hack.codingblocks.com/app/practice/1/282/problem
 */
public class Main {
	// My Input Function Taking irregular Testcases as input
	static String takeInput(BufferedReader sc, int n) throws Exception {
		String str = "";
		while(str.split(" ").length <= n) str += " " + sc.readLine().trim().replaceAll("\\s+", " ");
		return str.trim();
	}
	static void r3Pattern2(int n) {
		int sum = 1;
		for(int row = 1; row <= 2 * n - 1; row++){
            if(row % 2 == 1){
                for(int col = 1; col <= sum; col++) System.out.print("*");
                sum++;
            }
            System.out.println();
        }
	}
	public static void main(String[] args) throws Exception {
		BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(takeInput(sc, 1));
        r3Pattern2(n);
        sc.close();
    }
}

File: src/r3_pattern2/main.py
'''
 @author: Pranshu Aggarwal
 @problem: https://hack.codingblocks.com/app/practice/1/282/problem
'''
import re

def r3_pattern2(n):
    sum = 1
    for row in range(1, 2 * n):
        if row % 2 == 1:
            for col in range(1, sum + 1): print("*", end='')
            sum += 1
        print()

# Taking irregular testcases like a Scanner Class in java
def take_input(n):
    arr = []
    while True:
        if len(arr) == n: break
        line = input().strip()
        re.sub(' +', ' ', line)
        arr.extend(line.split())
    return " ".join(arr).strip()

if __name__ == "__main__":
    n = int(take_input(1))
    r3_pattern2(n)